File: CabBooking/Controller.cs/RiderController.cs
using CabBooking.DataAccessLayer;
using CabBooking.Model;

namespace CabBooking.Controller
{
    class RiderController
    {
        RiderManager _riderManager;
        TripManager _tripManager;
        public RiderController(RiderManager riderManager, TripManager tripManager)
        {
            _riderManager = riderManager;
            _tripManager = tripManager;
        }
        public bool Register(string id, string mobile)
        {
            try
            {
                _riderManager.RegisterRider(id, mobile);
            }
            catch (Exception exception)
            {
                return false;
            }
            return true;
        }
        public void BookCab(string riderId, Location source, Location destination)
        {
            _tripManager.CreateTrip(_riderManager.Get(riderId), source, destination);
        }
        public List<Trip> GetTripsHistory(string riderId)
        {
            return _tripManager.GetRiderTripsHistory(riderId);
        }
    }
}

File: CabBooking/Cab Booking Java/src/main/java/org/example/controllers/RiderController.java
package org.example.controllers;

import lombok.AllArgsConstructor;
import org.example.model.DriverStatus;
import org.example.model.Location;
import org.example.model.Ride;
import org.example.service.RideService;
import org.example.service.RiderService;

import java.util.List;
import java.util.Objects;

@AllArgsConstructor
public class RiderController {

    private final RiderService riderService;

    private final RideService rideService;

    public String register(String phoneNumber) {

        return riderService.register(phoneNumber);
    }

    public void bookCab(String riderId, Location source, Location destination) {

        riderService.bookCab(riderId, source, destination);
    }

    public List<Ride> fetchRideHistory(String riderId) {
        return rideService.fetchRideHistory(riderId);
    }
}
